src/custom/float.rs

```diff
@@ -89,20 +94,83 @@ impl<'c, H: HugrView> CodegenExtension<'c, H> for FloatOpsCodegenExtension {
+            "",
+        )?;
+        // convert to whatever BOOL_T is
+        Ok(vec![builder.build_select(r, true_val, false_val, "")?])
+    })
+}
+
 impl<'c, H: HugrView> EmitOp<'c, CustomOp, H> for FloatOpEmitter<'c, '_, H> {
     fn emit(&mut self, args: EmitOpArgs<'c, CustomOp, H>) -> Result<()> {
-        use hugr::ops::NamedOp;
-        let name = args.node().name();
-        // This looks strange now, but we will add cases for ops piecemeal, as
-        // in the analgous match expression in `IntOpEmitter`.
-        #[allow(clippy::match_single_binding)]
-        match name.as_str() {
-            n => Err(anyhow!("FloatOpEmitter: unknown op: {n}")),
+        let op = FloatOps::from_optype(&args.node().generalise()).ok_or(anyhow!(
+            "LogicOpEmitter: from_optype_failed: {:?}",
+            args.node().as_ref()
+        ))?;
+        // We emit the float comparison variants where NaN is an absorbing value.
+        // Any comparison with NaN is always false.
+        #[allow(clippy::wildcard_in_or_patterns)]
+        match op {
+            FloatOps::feq => emit_fcmp(self.0, args, inkwell::FloatPredicate::OEQ),
+            FloatOps::fne => emit_fcmp(self.0, args, inkwell::FloatPredicate::ONE),
+            FloatOps::flt => emit_fcmp(self.0, args, inkwell::FloatPredicate::OLT),
+            FloatOps::fgt => emit_fcmp(self.0, args, inkwell::FloatPredicate::OGT),
+            FloatOps::fle => emit_fcmp(self.0, args, inkwell::FloatPredicate::OLE),
+            FloatOps::fge => emit_fcmp(self.0, args, inkwell::FloatPredicate::OGE),
+            FloatOps::fadd => emit_custom_binary_op(self.0, args, |builder, (lhs, rhs), _| {
+                Ok(vec![builder
+                    .build_float_add(lhs.into_float_value(), rhs.into_float_value(), "")?
+                    .as_basic_value_enum()])
+            }),
+            FloatOps::fsub => emit_custom_binary_op(self.0, args, |builder, (lhs, rhs), _| {
+                Ok(vec![builder
+                    .build_float_sub(lhs.into_float_value(), rhs.into_float_value(), "")?
+                    .as_basic_value_enum()])
+            }),
+            FloatOps::fneg => emit_custom_unary_op(self.0, args, |builder, v, _| {
+                Ok(vec![builder
+                    .build_float_neg(v.into_float_value(), "")?
+                    .as_basic_value_enum()])
+            }),
+            FloatOps::fmul => emit_custom_binary_op(self.0, args, |builder, (lhs, rhs), _| {
+                Ok(vec![builder
+                    .build_float_mul(lhs.into_float_value(), rhs.into_float_value(), "")?
+                    .as_basic_value_enum()])
+            }),
+            FloatOps::fdiv => emit_custom_binary_op(self.0, args, |builder, (lhs, rhs), _| {
+                Ok(vec![builder
+                    .build_float_div(lhs.into_float_value(), rhs.into_float_value(), "")?
+                    .as_basic_value_enum()])
+            }),
+            // Missing ops, not supported by inkwell
```

Comment:
Can you create an issue for the missing ops?

Reply:
Will do